pkg/controller/workspace/workspace_controller.go

```diff
@@ -137,8 +137,9 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 }
 
 func (r *Reconciler) bindWorkspace(ctx context.Context, logger logr.Logger, namespace *corev1.Namespace, workspace *tenantv1alpha1.Workspace) error {
+	kubefedManaged := namespace.Labels[constants.KubefedManagedLabel] == "true"
 	// owner reference not match workspace label
-	if !metav1.IsControlledBy(namespace, workspace) {
+	if !metav1.IsControlledBy(namespace, workspace) && !kubefedManaged {
```

Comment:
Why do we check the kubefedManaged again? We have checked that before the function call.